src/components/overflow-menu/overflow-menu.js

```diff
@@ -156,17 +162,24 @@ class OverflowMenu extends mixin(createComponent, initComponentBySearch, evented
    */
   _handleKeyPress(event) {
     const key = event.which;
-    if (key === 13) {
-      const { element, optionMenu, options } = this;
+    const { element, optionMenu, options } = this;
+    const isOfMenu = optionMenu && optionMenu.element.contains(event.target);
+
+    if (key === 27) {
+      this.changeState('hidden', getLaunchingDetails(event), () => {
+        if (isOfMenu) {
+          element.focus();
+        }
+      });
+    }
+
+    if (key === 13 || key === 32) {
       const isOfSelf = element.contains(event.target);
-      const isOfMenu = optionMenu && optionMenu.element.contains(event.target);
       const shouldBeOpen = isOfSelf && !element.classList.contains(options.classShown);
       const state = shouldBeOpen ? 'shown' : 'hidden';
 
       if (isOfSelf) {
-        if (element.tagName === 'A') {
```

Comment:
The code here is for preventing link from being followed. Did you find a new condition where event's default behavior should be prevented?

Reply:
I was running into issues when trying to open the menu with the spacebar. It was jumping the page. Perhaps I should I only prevent that if the key === space?

Reply:
Let me know if this is a better solution

Reply:
Thanks @tw15egan for the background and the suggestion! What I saw is that the change from `keypress` to `keydown` caused the overflow menu to be open (and thus focus moving to the overflow menu item) _in the middle of_ the keystroke (of enter key). When user finishes the keystroke, the `keypress` event fires on the overflow menu item given it's a `<button>` and now has focus.

One way to get around with this is what you suggested. Great to have above finding as comments so that future maintainers/consumers can figure out why, if we go with this approach.

Another approach to address this is below, which directly removes the weird behavior described above, and thus reduces possibility of future issues (esp. wrt browser compatibility).

```diff
diff --git a/src/components/overflow-menu/overflow-menu.js b/src/components/overflow-menu/overflow-menu.js
index bdc031f..4a7ffc2 100644
--- a/src/components/overflow-menu/overflow-menu.js
+++ b/src/components/overflow-menu/overflow-menu.js
@@ -82,7 +82,16 @@ class OverflowMenu extends mixin(createComponent, initComponentBySearch, evented
     );
     this.manage(
       on(this.element.ownerDocument, 'keydown', event => {
-        this._handleKeyPress(event);
+        if (event.which === 27) {
+          this._handleKeyPress(event);
+        }
+      })
+    );
+    this.manage(
+      on(this.element.ownerDocument, 'keypress', event => {
+        if (event.which !== 27) {
+          this._handleKeyPress(event);
+        }
       })
     );
     this.manage(
@@ -179,7 +188,7 @@ class OverflowMenu extends mixin(createComponent, initComponentBySearch, evented
       const state = shouldBeOpen ? 'shown' : 'hidden';

       if (isOfSelf) {
-        if (element.tagName === 'A' || key === 13) {
+        if (element.tagName === 'A') {
           event.preventDefault();
         }
         event.delegateTarget = element; // eslint-disable-line no-param-reassign
```

Don't hesitate to ping me if you have any questions/concerns. Thanks!

Reply:
@asudoh made the changes, but still found the need to prevent default when the space key is pressed (https://github.com/IBM/carbon-components/pull/1115/files#diff-cdb22c353cb80d123893d9f188477d1bR191) to prevent the screen from jumping down

Reply:
Thanks alot @tw15egan! Could you add a comment to describing `32` key default prevention is for preventing screen from jumping down? I think this PR is good to go once it happens - Thanks!

Reply:
Added the comment in, going to merge it! 😄 Thanks for helping out